os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -111,9 +112,19 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
 
 		mm_takesemaphore(heap);
 
-		for (node = heap->mm_heapstart[region]; node < heap->mm_heapend[region]; node = (struct mm_allocnode_s *)((char *)node + node->size)) {
+		prev = NULL;
+		for (node = heap->mm_heapstart[region]; node < heap->mm_heapend[region]; prev = node, node = (struct mm_allocnode_s *)((char *)node + node->size)) {
 			mvdbg("region=%d node=%p size=%p preceding=%p (%c)\n", region, node, node->size, (node->preceding & ~MM_ALLOC_BIT), (node->preceding & MM_ALLOC_BIT) ? 'A' : 'F');
 
+			if (prev && (prev->size != (node->preceding & ~MM_ALLOC_BIT))) {
+				mdbg("ERROR: Heap node corruption detected\n");
+				mdbg("Corrupted node addr = 0x%08x size = %u\n", prev, prev->size);
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+				mdbg("Corrupted node owner pid = %u, alloc_call_addr = 0x%08x\n", prev->pid, prev->alloc_call_addr);
+#endif
+				DEBUGASSERT(0);
```

Comment:
Is debugassert enough? Because our mm is broken, we can run mm normally. So I think it should be assert, not debug assert.

Reply:
In my opinion, we can use debug assert when it does not affect system operation even it's wrong. Then we can debug that with config_debug enabled.

Reply:
True. I had a confusion about this. I have seen some places in code, where debugassert is used even in critical case. 
Let me change this to assert.